### PR TITLE
fix(cat-voices): update key derivation path

### PR DESCRIPTION
**Pull request by dtscalac**
# Description

Updates key derivation path based on new specification: https://github.com/input-output-hk/catalyst-voices/pull/1300

## Please confirm the following checks

* [x] My code follows the style guidelines of this project
* [x] I have performed a self-review of my code
* [x] I have commented my code, particularly in hard-to-understand areas
* [x] I have made corresponding changes to the documentation
* [x] My changes generate no new warnings
* [x] I have added tests that prove my fix is effective or that my feature works
* [x] New and existing unit tests pass locally with my changes
* [x] Any dependent changes have been merged and published in downstream module

### CODE REVIEW

File: catalyst_voices/packages/internal/catalyst_voices_models/lib/src/user/account_role.dart [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
+  drep(roleNumber: 1),
 
-  // TODO(dtscalac): the RBAC specification doesn't define yet the role number
-  // for the drep, replace this arbitrary number when it's specified.
-  drep(roleNumber: 2);
+  /// An account role that can create new proposals.
+  proposer(roleNumber: 3);
 
   /// The RBAC specified role number.
   final int roleNumber;
```
[the remark this answers is not in the document]
Comment (dtscalac) [reply catalyst_voices/packages/internal/catalyst_voices_models/lib/src/user/account_role.dart:13]: done

File: catalyst_voices/packages/internal/catalyst_voices_services/lib/src/crypto/key_derivation.dart [hunk 2]
```diff
@@ -45,8 +45,8 @@ final class KeyDerivation {
   /// The path feed into key derivation algorithm
   /// to generate a key pair from a seed phrase.
   ///
-  // TODO(dtscalac): update when RBAC specifies it
+  /// See: https://github.com/input-output-hk/catalyst-voices/pull/1300
   String _roleKeyDerivationPath(AccountRole role) {
-    return "m/${role.roleNumber}'/1234'";
+    return "m/508'/139'/0'/${role.roleNumber}/0";
```
[the remark this answers is not in the document]
Comment (dtscalac) [reply catalyst_voices/packages/internal/catalyst_voices_services/lib/src/crypto/key_derivation.dart:50]: done

**Review by damian-molinski — APPROVED** [review]
LGTM

File: catalyst_voices/packages/internal/catalyst_voices_services/lib/src/crypto/key_derivation.dart [hunk 3]
```diff
@@ -3,6 +3,10 @@ import 'package:catalyst_voices_models/catalyst_voices_models.dart';
 
 /// Derives key pairs from a seed phrase.
 final class KeyDerivation {
+  /// See: https://github.com/input-output-hk/catalyst-voices/pull/1300
+  static const int _purpose = 508;
+  static const int _type = 139;
+
```
Comment (stevenj) [inline catalyst_voices/packages/internal/catalyst_voices_services/lib/src/crypto/key_derivation.dart:9]: Just for alignment with the spec,  I suggest adding this.
```suggestion
static const int _account = 0; // Future Use
```

File: catalyst_voices/packages/internal/catalyst_voices_services/lib/src/crypto/key_derivation.dart [hunk 4]
```diff
@@ -44,9 +48,7 @@ final class KeyDerivation {
 
   /// The path feed into key derivation algorithm
   /// to generate a key pair from a seed phrase.
-  ///
-  // TODO(dtscalac): update when RBAC specifies it
   String _roleKeyDerivationPath(AccountRole role) {
-    return "m/${role.roleNumber}'/1234'";
+    return "m/$_purpose'/$_type'/0'/${role.number}/0";
```
Comment (stevenj) [inline catalyst_voices/packages/internal/catalyst_voices_services/lib/src/crypto/key_derivation.dart:52]:
```suggestion
    return "m/$_purpose'/$_type'/$_account'/${role.number}/0";
```

**Review by stevenj — CHANGES_REQUESTED** [review]
Changes are not required, just a suggestion.
Otherwise LGTM

**Review by damian-molinski — APPROVED** [review]

**Review by stevenj — APPROVED** [review]
LGTM

**Pull request merged**